File: framework/languages/java/e2e/RuntimeMonitoring/Server/Service/src/main/java/systems/zlink/e2e/runtimemonitoring/service/handlers/MonitoringActor.java
package systems.zlink.e2e.runtimemonitoring.service.handlers;

import systems.zlink.framework.actors.ZLinkActor;
import systems.zlink.framework.actors.ZLinkActorContext;

public final class MonitoringActor implements ZLinkActor {
    public static final String TYPE = "monitoring-placement-actor";
    private final ZLinkActorContext context;

    public MonitoringActor(ZLinkActorContext context) {
        this.context = context;
    }

    @Override
    public ZLinkActorContext context() {
        return context;
    }
}

File: framework/languages/java/e2e-kotlin/RuntimeMonitoring/Server/Service/src/main/kotlin/systems/zlink/e2e/kotlin/runtimemonitoring/service/MonitoringActor.kt
package systems.zlink.e2e.kotlin.runtimemonitoring.service

import systems.zlink.framework.actors.ZLinkActor
import systems.zlink.framework.actors.ZLinkActorContext
import systems.zlink.framework.kotlin.ZLinkSuspendingActorFactory

class MonitoringActor(
    private val id: String,
    private val actorContext: ZLinkActorContext,
) : ZLinkActor {
    override fun context(): ZLinkActorContext = actorContext
    fun actorId(): String = id
}

class MonitoringActorFactory : ZLinkSuspendingActorFactory() {
    override suspend fun createActor(context: ZLinkActorContext): ZLinkActor =
        MonitoringActor(context.actorId(), context)
}
